proto/github.proto

```diff
@@ -21,3 +21,170 @@ message UnlinkGitHubAccountResponse {
+
+  // The GitHub username or organization where the app is installed.
+  // This will be unique for each installation.
+  string owner = 2;
+
+  // The group ID to which the installation is linked.
+  string group_id = 3;
+}
+
+// A request to list GitHub App installations accessible to user as well as
+// installations linked to the org (which may or may not be directly accessible
+// to the user).
+//
+// Installations are returned for all groups that are accessible to the user, so
+// that the user can get a clear view of which installations are linked to which
+// orgs, without needing to switch orgs.
+message GetAppInstallationsRequest {
+  context.RequestContext request_context = 1;
+}
+
+message GetAppInstallationsResponse {
+  context.ResponseContext response_context = 1;
+
+  // The installations available to the user.
+  repeated AppInstallation installations = 2;
+}
+
+// A request to update a GitHub App installation. The authenticated user must
+// have a linked GitHub account which has access to the installation.
+message UpdateAppInstallationRequest {
+  context.RequestContext request_context = 1;
+
+  // The installation ID to update.
+  // Required.
+  int64 installation_id = 2;
+
+  // The group to assign or re-assign the installation to.
+  // Required.
+  string group_id = 3;
+}
+
+message UpdateAppInstallationResponse {
+  context.ResponseContext response_context = 1;
+}
+
+// A request to unlink a GitHub App installation from the authenticated org.
+//
+// The authenticated user does not have to have access to the installation in
+// order to unlink it from the org.
+//
+// Unlinking an installation does not automatically unlink all repositories
+// associated with the installation. However, members of the organization will
+// no longer be able to get repository access tokens (unless the installation is
+// re-linked).
+message UnlinkAppInstallationRequest {
+  context.RequestContext request_context = 1;
+
+  // The installation ID to unlink.
+  int64 installation_id = 2;
+}
+
+message UnlinkAppInstallationResponse {
+  context.ResponseContext response_context = 1;
+}
+
+// A request to list repos available to a particular installation.
+message GetReposRequest {
+  context.RequestContext request_context = 1;
+
+  // The GitHub username or organization name to search within. The
+  // authenticated user must have access to a GitHub App installation associated
+  // with the repo
+  string owner = 2;
+
+  // The search query (optional). If no query is specified, list all
+  // repositories.
+  string query = 3;
+}
+
+message GetReposResponse {
+  context.ResponseContext response_context = 1;
+
+  message Repo {
+    // The repo URL.
+    // Example: "https://github.com/buildbuddy-io/buildbuddy"
+    string repo_url = 1;
+
+    // Whether the repository is linked to the authenticated organization.
+    bool linked = 2;
+  }
+
+  // The repos matching the request criteria.
+  repeated Repo repos = 2;
```

Comment:
we have a few random things that violate this, but generally speaking the proto convention is to use singular nouns even for repeated fields.   pretty sure this convention arises from how lists are accessed in some languages.

https://github.com/search?q=repo%3Abuildbuddy-io%2Fbuildbuddy+%22repeated+%22&type=code&p=2

Reply:
Fixed

Reply:
lol, it looks like I didn't actually fix it in my original commit. I will just keep it as-is then :sweat_smile: